### PR TITLE
added new function "findApiChanges" to find diffs and apply rules

### PR DESCRIPTION
**Pull request by unandyala**

### CODE REVIEW

File: src/diffTool/diffProcessor.ts [hunk 1]
```diff
@@ -14,18 +14,37 @@ import { applyRules } from "./rulesProcessor";
...
+ */
+export async function findApiChanges(
+  leftRaml: string,
+  rightRaml: string,
+  rulesPath?: string
+): Promise<NodeDiff[]> {
+  const diffs = await diffRaml(leftRaml, rightRaml);
+  if (rulesPath == null) {
+    ramlToolLogger.info("Applying default rules on the differences");
+    rulesPath = path.join(diffRulesPath, "defaultRules.json");
```
Comment (wjhsf) [inline src/diffTool/diffProcessor.ts:33]: `diffRulesPath` is a constant, and is only used here. Make the filename part of that so you don't have to recompute every time.

File: src/diffTool/diffProcessor.ts [hunk 2]
```diff
@@ -12,20 +12,42 @@ import { ramlToolLogger } from "../common/logger";
...
+ * @param leftRaml - Base RAML file to compare
+ * @param rightRaml - Other RAML file to compare with the left RAML to find differences
+ * @param rulesPath - Optional rules file to apply on the differences
+ *
+ * @returns Changes to the APIs
+ */
+export async function findApiChanges(
+  leftRaml: string,
+  rightRaml: string,
+  rulesPath?: string
```
Comment (monkey-jeff) [inline src/diffTool/diffProcessor.ts:31]: Why not `rulesPath = defaultRulesPath` or set it to the actual rule here?  Then you don't need the null check.

File: src/diffTool/diffProcessor.ts [hunk 3]
```diff
@@ -12,20 +12,42 @@ import { ramlToolLogger } from "../common/logger";
...
+
+/**
+ * Generate differences between two RAML files and apply rules
+ * @param leftRaml - Base RAML file to compare
+ * @param rightRaml - Other RAML file to compare with the left RAML to find differences
+ * @param rulesPath - Optional rules file to apply on the differences
+ *
+ * @returns Changes to the APIs
+ */
+export async function findApiChanges(
```
Comment (monkey-jeff) [inline src/diffTool/diffProcessor.ts:28]: Should this be `applyChangeRules` instead?

Comment (unandyala) [reply src/diffTool/diffProcessor.ts:28]: There is already function apply rules that applies the rules on node diffs. 
`applyRules(diffs: NodeDiff[],rulesPath: string)`

I thought we agreed on "findApiChanges". Once we have **ApiChanges** object, its return value will also be changed to **ApiChanges**

Comment (monkey-jeff) [reply src/diffTool/diffProcessor.ts:28]: you're right.  I retract my comment

**Pull request merged**